core/file_walker.go

```diff
@@ -40,17 +37,16 @@ func (f *File) UpdateSize() {
 
 type ReadDir func(dirname string) ([]os.FileInfo, error)
 
-func GetSubTree(path string, parent *File, readDir ReadDir, ignoredFolders map[string]struct{}) *File {
-	var mutex sync.Mutex
+func WalkFolder(path string, readDir ReadDir, ignoredFolders map[string]struct{}) *File {
 	var wg sync.WaitGroup
-	c := make(chan bool, maxConcurrentScans)
-	root := getSubTreeConcurrently(path, parent, readDir, ignoredFolders, c, &mutex, &wg)
+	c := make(chan bool, runtime.NumCPU())
+	root := getSubTreeConcurrently(path, nil, readDir, ignoredFolders, c, &wg)
 	wg.Wait()
 	root.UpdateSize()
 	return root
 }
 
-func getSubTreeConcurrently(path string, parent *File, readDir ReadDir, ignoredFolders map[string]struct{}, c chan bool, mutex *sync.Mutex, wg *sync.WaitGroup) *File {
+func getSubTreeConcurrently(path string, parent *File, readDir ReadDir, ignoredFolders map[string]struct{}, c chan bool, wg *sync.WaitGroup) *File {
```

Comment:
This function should be renamed to something else. Go funcs usually don't start with "Get"